File: backend-ms1/src/main/java/ma/zs/zyn/service/facade/admin/recomrecla/RecommendTeacherAdminService.java
package ma.zs.zyn.service.facade.admin.recomrecla;

import java.util.List;
import ma.zs.zyn.bean.core.recomrecla.RecommendTeacher;
import ma.zs.zyn.dao.criteria.core.recomrecla.RecommendTeacherCriteria;
import ma.zs.zyn.zynerator.service.IService;



public interface RecommendTeacherAdminService extends  IService<RecommendTeacher,RecommendTeacherCriteria>  {

    List<RecommendTeacher> findByProfId(Long id);
    int deleteByProfId(Long id);
    long countByProfRef(String ref);




}

File: frontend/src/app/shared/service/admin/recomrecla/RecommendTeacherAdmin.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from '@angular/common/http';
import {BehaviorSubject, Observable} from 'rxjs';

import {environment} from 'src/environments/environment';

import {RecommendTeacherDto} from 'src/app/shared/model/recomrecla/RecommendTeacher.model';
import {RecommendTeacherCriteria} from 'src/app/shared/criteria/recomrecla/RecommendTeacherCriteria.model';
import {AbstractService} from 'src/app/zynerator/service/AbstractService';


@Injectable({
  providedIn: 'root'
})
export class RecommendTeacherAdminService extends AbstractService<RecommendTeacherDto, RecommendTeacherCriteria> {
     constructor(private http: HttpClient) {
        super();
        this.setHttp(http);
    }

    public constrcutDto(): RecommendTeacherDto {
        return new RecommendTeacherDto();
    }

    public constrcutCriteria(): RecommendTeacherCriteria {
        return new RecommendTeacherCriteria();
    }

    get API() {
        return environment.apiUrlZynservice + 'admin/recommendTeacher/';
    }
}
